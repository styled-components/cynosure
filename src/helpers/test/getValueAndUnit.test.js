// @flow
import getValueAndUnit from '../getValueAndUnit'

describe('getValueAndUnit', () => {
  it('should get value and px from whole values', () => {
    expect(getValueAndUnit('1px')).toMatchSnapshot()
  })

  it('should get value and px from values', () => {
    expect(getValueAndUnit('1.5px')).toMatchSnapshot()
  })

  it('should get value and pt from whole values', () => {
    expect(getValueAndUnit('1pt')).toMatchSnapshot()
  })

  it('should get value and pt from values', () => {
    expect(getValueAndUnit('1.5pt')).toMatchSnapshot()
  })

  it('should get value and pc from whole values', () => {
    expect(getValueAndUnit('1pc')).toMatchSnapshot()
  })

  it('should get value and pc from values', () => {
    expect(getValueAndUnit('1.5pc')).toMatchSnapshot()
  })

  it('should get value and mm from whole values', () => {
    expect(getValueAndUnit('1mm')).toMatchSnapshot()
  })

  it('should get value and mm from values', () => {
    expect(getValueAndUnit('1.5mm')).toMatchSnapshot()
  })

  it('should get value and q from whole values', () => {
    expect(getValueAndUnit('1q')).toMatchSnapshot()
  })

  it('should get value and q from values', () => {
    expect(getValueAndUnit('1.5q')).toMatchSnapshot()
  })

  it('should get value and cm from whole values', () => {
    expect(getValueAndUnit('1cm')).toMatchSnapshot()
  })

  it('should get value and cm from values', () => {
    expect(getValueAndUnit('1.5cm')).toMatchSnapshot()
  })

  it('should get value and in from whole values', () => {
    expect(getValueAndUnit('1in')).toMatchSnapshot()
  })

  it('should get value and in from values', () => {
    expect(getValueAndUnit('1.5in')).toMatchSnapshot()
  })

  it('should get value and em from whole value', () => {
    expect(getValueAndUnit('1em')).toMatchSnapshot()
  })

  it('should get value and em from decimal values', () => {
    expect(getValueAndUnit('1.2em')).toMatchSnapshot()
  })

  it('should get value and rem from whole values', () => {
    expect(getValueAndUnit('1rem')).toMatchSnapshot()
  })

  it('should get value and rem from decimal values', () => {
    expect(getValueAndUnit('1.2rem')).toMatchSnapshot()
  })

  it('should get value and ex from whole values', () => {
    expect(getValueAndUnit('1ex')).toMatchSnapshot()
  })

  it('should get value and ex from decimal values', () => {
    expect(getValueAndUnit('1.2ex')).toMatchSnapshot()
  })

  it('should get value and ch from whole values', () => {
    expect(getValueAndUnit('1ch')).toMatchSnapshot()
  })

  it('should get value and ch from decimal values', () => {
    expect(getValueAndUnit('1.2ch')).toMatchSnapshot()
  })

  it('should get value and vh from whole values', () => {
    expect(getValueAndUnit('100vh')).toMatchSnapshot()
  })

  it('should get value and vh from decimal values', () => {
    expect(getValueAndUnit('33.33vh')).toMatchSnapshot()
  })

  it('should get value and vw from whole values', () => {
    expect(getValueAndUnit('100vw')).toMatchSnapshot()
  })

  it('should get value and vw from decimal values', () => {
    expect(getValueAndUnit('33.33vw')).toMatchSnapshot()
  })

  it('should get value and vmin from whole values', () => {
    expect(getValueAndUnit('100vmin')).toMatchSnapshot()
  })

  it('should get value and vmin from decimal values', () => {
    expect(getValueAndUnit('33.33vmin')).toMatchSnapshot()
  })

  it('should get value and vmax from whole values', () => {
    expect(getValueAndUnit('100vmax')).toMatchSnapshot()
  })

  it('should get value and vmax from decimal values', () => {
    expect(getValueAndUnit('33.33vmax')).toMatchSnapshot()
  })

  it('should get value and % from whole values', () => {
    expect(getValueAndUnit('80%')).toMatchSnapshot()
  })

  it('should get value and % from decimal values', () => {
    expect(getValueAndUnit('33.3%')).toMatchSnapshot()
  })
})
