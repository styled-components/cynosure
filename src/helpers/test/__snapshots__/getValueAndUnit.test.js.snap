// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getValueAndUnit should get value and % from decimal values 1`] = `
Array [
  33.3,
  "%",
]
`;

exports[`getValueAndUnit should get value and % from whole values 1`] = `
Array [
  80,
  "%",
]
`;

exports[`getValueAndUnit should get value and ch from decimal values 1`] = `
Array [
  1.2,
  "ch",
]
`;

exports[`getValueAndUnit should get value and ch from whole values 1`] = `
Array [
  1,
  "ch",
]
`;

exports[`getValueAndUnit should get value and cm from values 1`] = `
Array [
  1.5,
  "cm",
]
`;

exports[`getValueAndUnit should get value and cm from whole values 1`] = `
Array [
  1,
  "cm",
]
`;

exports[`getValueAndUnit should get value and em from decimal values 1`] = `
Array [
  1.2,
  "em",
]
`;

exports[`getValueAndUnit should get value and em from whole value 1`] = `
Array [
  1,
  "em",
]
`;

exports[`getValueAndUnit should get value and ex from decimal values 1`] = `
Array [
  1.2,
  "ex",
]
`;

exports[`getValueAndUnit should get value and ex from whole values 1`] = `
Array [
  1,
  "ex",
]
`;

exports[`getValueAndUnit should get value and in from values 1`] = `
Array [
  1.5,
  "in",
]
`;

exports[`getValueAndUnit should get value and in from whole values 1`] = `
Array [
  1,
  "in",
]
`;

exports[`getValueAndUnit should get value and mm from values 1`] = `
Array [
  1.5,
  "mm",
]
`;

exports[`getValueAndUnit should get value and mm from whole values 1`] = `
Array [
  1,
  "mm",
]
`;

exports[`getValueAndUnit should get value and pc from values 1`] = `
Array [
  1.5,
  "pc",
]
`;

exports[`getValueAndUnit should get value and pc from whole values 1`] = `
Array [
  1,
  "pc",
]
`;

exports[`getValueAndUnit should get value and pt from values 1`] = `
Array [
  1.5,
  "pt",
]
`;

exports[`getValueAndUnit should get value and pt from whole values 1`] = `
Array [
  1,
  "pt",
]
`;

exports[`getValueAndUnit should get value and px from values 1`] = `
Array [
  1.5,
  "px",
]
`;

exports[`getValueAndUnit should get value and px from whole values 1`] = `
Array [
  1,
  "px",
]
`;

exports[`getValueAndUnit should get value and q from values 1`] = `
Array [
  1.5,
  "q",
]
`;

exports[`getValueAndUnit should get value and q from whole values 1`] = `
Array [
  1,
  "q",
]
`;

exports[`getValueAndUnit should get value and rem from decimal values 1`] = `
Array [
  1.2,
  "rem",
]
`;

exports[`getValueAndUnit should get value and rem from whole values 1`] = `
Array [
  1,
  "rem",
]
`;

exports[`getValueAndUnit should get value and vh from decimal values 1`] = `
Array [
  33.33,
  "vh",
]
`;

exports[`getValueAndUnit should get value and vh from whole values 1`] = `
Array [
  100,
  "vh",
]
`;

exports[`getValueAndUnit should get value and vmax from decimal values 1`] = `
Array [
  33.33,
  "vmax",
]
`;

exports[`getValueAndUnit should get value and vmax from whole values 1`] = `
Array [
  100,
  "vmax",
]
`;

exports[`getValueAndUnit should get value and vmin from decimal values 1`] = `
Array [
  33.33,
  "vmin",
]
`;

exports[`getValueAndUnit should get value and vmin from whole values 1`] = `
Array [
  100,
  "vmin",
]
`;

exports[`getValueAndUnit should get value and vw from decimal values 1`] = `
Array [
  33.33,
  "vw",
]
`;

exports[`getValueAndUnit should get value and vw from whole values 1`] = `
Array [
  100,
  "vw",
]
`;
